src/role.ts

```diff
@@ -112,6 +159,20 @@ export class GithubActionsRole extends iam.Role {
     }
   }
 
+  /** Validates conflicting props aren't set simultaneously */
+  private static validateProps(scope: Construct, props: GithubActionsRoleProps) {
+    const topLevelPropsSet = props.owner || props.repo || props.filter;
+    const trustedRepositoriesSet = props.trustedRepositories && props.trustedRepositories.length > 0 || false;
+
+    if (topLevelPropsSet && trustedRepositoriesSet) {
+      cdk.Annotations.of(scope).addError('Cannot set both top-level owner/repo/filter and trustedRepositories. Use one or the other.');
+    }
+
+    if (!trustedRepositoriesSet && (props.owner === undefined || props.repo === undefined)) {
+      cdk.Annotations.of(scope).addError("If you don't provide `trustedRepositories`, you must provide `owner` and `repo`.");
+    }
```

Comment:
If we decided to deprecate the top-level properties, we could annotate the deprecated properties here.